[CMake] rewrite-compile-commands reports a lot of warning messages "Failed to find ..."

fujii opened the pull request:
#### b332b310db505e67e0cc8561fcdb47bd3117b79a
<pre>
[CMake] rewrite-compile-commands reports a lot of warning messages &quot;Failed to find ...&quot;
<a href="https://bugs.webkit.org/show_bug.cgi?id=260841">https://bugs.webkit.org/show_bug.cgi?id=260841</a>

Reviewed by Don Olmstead.

rewrite-compile-commands scripts reported a lot of warning messages of
missing derived sources for the build-webkit first invocation. We
don&apos;t have a nice solution yet. And, derived sources don&apos;t have to be
included in compile_commands.json because we don&apos;t directly edit them.
Let&apos;s just ignore the missing derived sources at the moment.

* Tools/Scripts/rewrite-compile-commands:

Canonical link: <a href="https://commits.webkit.org/267542@main">https://commits.webkit.org/267542@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/344585be2cab1e8dd47fb2a58934b508954292cc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16889 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/17212 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17669 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18672 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15811 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/17079 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20444 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/17350 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/18069 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/17088 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/17447 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/14616 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19479 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14690 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/15308 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/22045 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/15675 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15475 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/19794 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/16088 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/13635 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/15254 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4050 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19617 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15918 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Tools/Scripts/rewrite-compile-commands:
```diff
@@ -86,7 +86,8 @@ for entry in compile_commands:
                     break
             else:
-                print('Failed to find', include_path)
+                # TODO: Don't just ignore derived source that is not generated yet.
```
Honestly, why not? Humans are not expected to edit these files, so is it really a big deal for them to be omitted?

mcatanzaro replied:
i.e. you can consider landing this without the TODO.

fujii replied:
I agree with you. Let's wait for @TingPing opinion.

TingPing replied:
It is something I'd like to handle eventually but the TODO isn't super valuable. I have no strong opinion.

fujii replied:
Fixed.

fujii committed:
[CMake] rewrite-compile-commands reports a lot of warning messages "Failed to find ..."
https://bugs.webkit.org/show_bug.cgi?id=260841

Reviewed by Don Olmstead.

rewrite-compile-commands scripts reported a lot of warning messages of
missing derived sources for the build-webkit first invocation. We
don't have a nice solution yet. And, derived sources don't have to be
included in compile_commands.json because we don't directly edit them.
Let's just ignore the missing derived sources at the moment.

* Tools/Scripts/rewrite-compile-commands:

Canonical link: https://commits.webkit.org/267542@main
